### ISSUE TITLE
Added Node is blinking

### ISSUE THREAD

**Issue by YoannBuzenet** [wbkd/react-flow#500]
Hello,

First thank you for this amazing library. I really like it, well done and thank you for this job !

I have a question. I'm implementing a diagram from the code of a collegue, and there is a problem when I test it : when adding a new node in the array of **elements**, the new node blinks : it appears and then disappears really quickly. But it's still in the array. We must be doing something wrong.

Here is all the code. I don't know if I could reproduce it in a simplier way on code sandbox (maybe it's dirty code that provokes this behaviour).

I apologize in advance, the code is very messy.

Thank you for your help, if you can read this :D
Yoann

```javascript
// Get items from API and transform them 
function transformCartoObjectFromServIntoDiagram(cartoData) {
    let arrayToReturn = [];
    let htmlStyle = "";
    let stateIcon = "";

    // Creating the tasks
    arrayToReturn = cartoData.listtask.map((task) => {
        switch (task.color) {
            case 0:
                stateIcon = "iconGrey";

                break;
            case 1:
            case 5:
                stateIcon = "iconGreen";

                break;
            case 2:
            case 4:
                stateIcon = "iconOrange";

                break;
            case 3:
            case 7:
                stateIcon = "iconRed";
                break;
            default:
                stateIcon = "iconGrey";
                break;
        }
let taskType = "";
        switch (task.state) {
            case 0:
                taskType = "headerInactiveTask";
                break;
            case 1:
                taskType = "headerActiveTask";
                break;

            case 3:
               taskType = "headerFinishedTask";
                break;
            case 2:
               taskType = "headerCancelledTask";
                           
                break;
            default: taskType = "toto";
        }

        htmlStyle = (
           
            <section className="Shape">
                <div className={taskType}>
                    <section className="classIcon">
                        <img
                          alt=""

                          className="diagram_icon"
                          src={picto06} /// /TODO changer pour le dpt
                        />
                    </section>
                </div>
                <div className="descTask">
                    <span>{task.name}</span>
                </div>
                <div id="footerTask">
                    <section className="clockIcon">
                    <i id={stateIcon} className="fa fa-thermometer-half fa-2x" aria-hidden="true" />;
                    </section>
                </div>
            </section>
           
        );

        return {
            id: task.idtask,
            data: {
                label: htmlStyle,
                ...task
            },
             position: { x: task.x_position, y: task.y_position }
        };
    });

    // Creating the links
    let arrayOfLinks = [];
    arrayOfLinks = cartoData.taskancestor.map((taskLink) => ({
        id: `e${taskLink.idtask_parent}-${taskLink.idtask_child}`,
        source: taskLink.idtask_parent,
        target: taskLink.idtask_child,
        animated: true,
        type: 'step',
        arrowHeadType: "arrow"
 
    }));

    console.log("links", arrayOfLinks);
    // Junction between both arrays
    arrayToReturn = [...arrayToReturn, ...arrayOfLinks];

    console.log("array with everything", arrayToReturn);

    return arrayToReturn;
}

// Doc on this diagram
// https://github.com/wbkd/react-flow#readme
const BasicFlow = ({ actions, currentTask, history, match, tasksModel }) => {
    const API_URL = process.env.REACT_APP_API_URL;
   
    const [elements, setElements] = useState([]);
    const [nodesDraggableState, setNodeDraggableState] = useState(false);
    const [nodesConnectableState, setNodesConnectableState] = useState(false);
    const [diagramMode, setDiagramMode] = useState("");
    const [firstLoad, setFirstLoad] = useState(true);
    const [previousClick, setPreviousClick] = useState(0);
    const [previousCall, setPreviousCall] = useState(0);
    const [searchText, setSearchText] = useState("");
    const [addID, setAddID] = useState(1);
   
    const onConnect = (params) => setElements((els) => addEdge({ ...params, animated: true, arrowHeadType: "arrow", type: "step" }, els));
   
    // get cartography in URL
    useEffect(() => {
        const url = `${API_URL}/CartoComplexe/GetById/${match.params.id}`;
        fetch(url, {
            method: "GET",
            headers: {
                Authorization: `Bearer ${localStorage.getItem("adal.idtoken")}`,
                "Content-type": "application/json; charset=utf-8"
            }
        })
            .then((resp) => {
                if (resp.status === 200) {
                    resp.json().then((elmt) => {
                        console.trace(transformCartoObjectFromServIntoDiagram(elmt));
                        setElements(transformCartoObjectFromServIntoDiagram(elmt));
                    });
                }
                else {
                    throw resp;
                }
            })
            .catch((err) => {
                console.log(err);
                // TODO Remove store notions
                // actions.setDisplayAlert(errorHandling.checkErrorAPIResponse(err, err.status, store.getState()?.userInfos));
            });
    }, [match.params.id]);
    function modifNodeDraggableState() {
        setNodeDraggableState(!nodesDraggableState);
    }

    function openForm(e, element) {
        const now = Date.now();
// pour palier au click present sur le stop drag on verifie que la position de la souris a bien bougé par rapport au denier double click
        if (e.timeStamp === previousCall) {
            return;
        }
        if (diagramMode !== "Edit" || (diagramMode === "Edit" && now - previousClick < 250)) {
            console.log("onClick diagram", element);
            actions.setCurrentTask(element.data);
            actions.setDisplayForm({
                formToDisplay: TaskFormNew,
                sidebar: "expand",
                isDisplayedForm: true,
                vault: {
                    postURL: `${API_URL}/AttachedFiles/PostTaskFile?CLASS=1&IDTASK=${element.data.idtask}`,
                    getURL: `${API_URL}/AttachedFiles/GetByTaskId/${element.data.idtask}`
                },
                class: element.data.class,
                mode: match.params.mode // TODO PRECISE REAL MODE
            });
        }
        setPreviousClick(now);
        setPreviousCall(e.timeStamp);
    }
   
    function setViewMode() {
            setDiagramMode("View");
            setNodeDraggableState(false);
            document.getElementById("diagram.editButton").style.display = "block";
            document.getElementById("diagram.reinitializeButton").style.display = "block";
            document.getElementById("diagram.cancelButton").style.display = "none";
            document.getElementById("diagram.saveButton").style.display = "none";
            document.getElementsByClassName("react-flow")[0].style.backgroundImage = "none";
              /*eslint-disable */
            document.getElementsByClassName("react-flow")[0].style.width=defaultStatus
             /* eslint-enable */
            setNodesConnectableState(false);
        }
        function setEditMode() {
            setDiagramMode("Edit");
            setNodeDraggableState(true);
            setNodesConnectableState(true);
            document.getElementById("diagram.editButton").style.display = "none";
            document.getElementById("diagram.reinitializeButton").style.display = "none";
            document.getElementById("diagram.cancelButton").style.display = "block";
            document.getElementById("diagram.saveButton").style.display = "block";

              /*eslint-disable */
              document.getElementsByClassName("react-flow")[0].style.width="calc(100% - 250px)"
            document.getElementsByClassName("react-flow")[0].style.backgroundImage = defaultStatus;
              /* eslint-enable */
        }

    // Automatic dezoom to see the complete diagram
    const onLoad = (reactFlowInstance) => {
        reactFlowInstance.fitView();
        if (firstLoad) {
            actions.getTaskList(match.params.area);
            switch (match.params.mode) {
                case "c": setEditMode();
        
                break;
                default: setViewMode();
        
                break;
            }
            setFirstLoad(false);
        }
    };
    const handleChange = (evt) => {
        setSearchText(evt.target.value);
        };
      
      const handleClickNewTask = (task) => {
          const x = 200;// Math.round(Math.random() * window.innerWidth);
          const y = 200;// Math.round(Math.random() * window.innerHeight);

          const htmlStyle = (
           
            <section className="Shape">
                <div className="headerTemporaryTask">
                    <section className="classIcon">
                        <img
                          alt=""

                          className="diagram_icon"
                          src={picto06} /// /TODO changer pour le dpt
                        />
                    </section>
                </div>
                <div className="descTask">
                    <span>{task.longname}</span>
                </div>
                <div id="footerTask" />
            </section>
           
        );

          const newNode = {
            id: addID,
            data: { id: `add-task-${addID}`,
            state: 0,
            label: htmlStyle,
            x,
            y,
            iddepartment: task.iddepartment,
            category: task.category
         },
            position: { x, y }
          };
         
          setElements([newNode, ...elements]);
          setAddID(addID + 1);
          console.trace(elements);
      };

    return (
        <>
            <header className="ContentHeader">
                <h1 className="Title">Folder  RENAULT Mégane Estate III 1.5 dCi 110 FAP eco2 Expression
                <input type="button" onClick={setEditMode} className="EditButton" id="diagram.editButton" defaultValue="Modifier" />
                <input type="button" className="EditButton" id="diagram.reinitializeButton" defaultValue="Réinitialiser la cartographie" />
                <input type="button" onClick={setViewMode} className="EditButton" id="diagram.cancelButton" defaultValue="Annuler" />
                <input type="button" className="EditButton" id="diagram.saveButton" defaultValue="Enregistrer" />
                </h1>
            </header>
            <ReactFlow
              elements={elements}
              elementsSelectable
              onLoad={onLoad}
              nodesDraggable={nodesDraggableState}
              paneMoveable
              zoomOnScroll
              arrowHeadColor="#206584"
              nodesConnectable={nodesConnectableState}
              defaultZoom={1}
              onElement
              onConnect={onConnect}
              onElementClick={openForm}
            >
              <MiniMap
                nodeColor={(node) => {
                switch (node.type) {
                  case 'input': return 'red';
                  case 'default': return '#00ff00';
                  case 'output': return 'rgb(0,0,255)';
                  default: return '#eee';
                }
              }}
              />
              <Controls />
            </ReactFlow>
            {diagramMode === 'Edit' && (
          <S.Toolbar>
            <S.Category>
              <S.CategoryTitle>
                <S.Search
                  type="text"
                  value={searchText}
                  onChange={handleChange}
                //  placeholder={translatedSearchTaskText}
                />
              </S.CategoryTitle>
              <S.CategoryContent>
                {tasksModel
                  && tasksModel
                    .filter((f) => f.longname.toLowerCase().includes(searchText.toLowerCase())
                  )
                    .map((t) => (
                      <S.CategoryButton
                        key={t.category}
                        onClick={() => handleClickNewTask(t)}
                      >
                        {t.longname}
                      </S.CategoryButton>
                    ))}
              </S.CategoryContent>
            </S.Category>
          </S.Toolbar>
        )}
        </>
    );
};

BasicFlow.propTypes = {
    actions: PropTypes.object.isRequired,
    currentTask: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired,
    id: PropTypes.string.isRequired,
    tasksModel: PropTypes.array

};
BasicFlow.defaultProps = { tasksModel: []

 };
export default withRouter(BasicFlow);
```

**Comment by moklick** [comment]
Hey. 

If it's not too complicated you could try to create a codesandbox, so that we can check the code. At first sight i can't find any error. We have an example where we are adding a node dynamically: [source](https://github.com/wbkd/react-flow/blob/master/example/src/Empty/index.js#L13-L21)  [demo](https://reactflow.dev/empty)

**Comment by YoannBuzenet** [comment]
We found the solution : the ID of the new node was an integer. If we pass it as a string everything works correctly.

**Comment by moklick** [comment]
Ok, great :) I will see if we can convert it or throw a warning if an integer is used.

**Comment by moklick** [comment]
A fix is released in v5.9.0
